cycles-ledger/tests/tests.rs

```diff
@@ -251,198 +501,157 @@ fn test_deposit_flow() {
 #[test]
 #[should_panic]
 fn test_deposit_amount_below_fee() {
-    let env = &new_state_machine();
-    let ledger_id = install_ledger(env);
-    let depositor_id = install_depositor(env, ledger_id);
-    let user = Account {
-        owner: Principal::from_slice(&[1]),
-        subaccount: None,
-    };
+    let env = TestEnv::setup();
+    let user = account(1, None);
 
     // Attempt to deposit fewer than [config::FEE] cycles. This call should panic.
-    let _deposit_result = deposit(env, depositor_id, user, config::FEE - 1, None);
+    let _deposit_result = env.deposit(user, config::FEE - 1, None);
 }
 
 #[test]
 fn test_withdraw_flow() {
     // TODO(SDK-1145): Add re-entrancy test
 
-    let env = &new_state_machine();
-    let ledger_id = install_ledger(env);
-    let depositor_id = install_depositor(env, ledger_id);
-    let user_main_account = Account {
-        owner: Principal::from_slice(&[1]),
-        subaccount: None,
-    };
-    let user_subaccount_1 = Account {
-        owner: Principal::from_slice(&[1]),
-        subaccount: Some([1; 32]),
-    };
-    let user_subaccount_2 = Account {
-        owner: Principal::from_slice(&[1]),
-        subaccount: Some([2; 32]),
-    };
-    let user_subaccount_3 = Account {
-        owner: Principal::from_slice(&[1]),
-        subaccount: Some([3; 32]),
-    };
-    let user_subaccount_4 = Account {
-        owner: Principal::from_slice(&[1]),
-        subaccount: Some([4; 32]),
-    };
-    let withdraw_receiver = env.create_canister(None);
+    let env = TestEnv::setup();
+    let user_main_account = account(1, None);
+    let user_subaccount_1 = account(1, Some(1));
+    let user_subaccount_2 = account(1, Some(2));
+    let user_subaccount_3 = account(1, Some(3));
+    let user_subaccount_4 = account(1, Some(4));
+    let withdraw_receiver = env.state_machine.create_canister(None);
 
     // make deposits to the user and check the result
-    let deposit_res = deposit(env, depositor_id, user_main_account, 1_000_000_000, None);
+    let deposit_res = env.deposit(user_main_account, 1_000_000_000, None);
     assert_eq!(deposit_res.block_index, 0_u128);
     assert_eq!(deposit_res.balance, 1_000_000_000_u128);
-    deposit(env, depositor_id, user_subaccount_1, 1_000_000_000, None);
-    deposit(env, depositor_id, user_subaccount_2, 1_000_000_000, None);
-    deposit(env, depositor_id, user_subaccount_3, 1_000_000_000, None);
-    deposit(env, depositor_id, user_subaccount_4, 1_000_000_000, None);
+    let _deposit_res = env.deposit(user_subaccount_1, 1_000_000_000, None);
```

Comment:
Deposit doesn't return a Result so it doesn't make sense to trap at that level. Note that it does panic already if the call can't reach the network or there is a reject from the network because it uses `update_or_trap` to make the call.